src/coreclr-debug/util.ts

```diff
@@ -117,3 +118,61 @@ export class CoreClrDebugUtil {
         return '';
     }
 }
+
+const MINIMUM_SUPPORTED_OSX_ARM64_DOTNET_CLI: string = '6.0.0';
+
+export function getTargetArchitecture(platformInfo: PlatformInformation, launchJsonTargetArchitecture: string, dotnetInfo: DotnetInfo): string 
+{
+    if (!platformInfo)
+    {
+        throw new Error(`Unable to retrieve 'TargetArchitecture' without platformInfo.`);
+    }
+
+    let targetArchitecture = "";
+
+    // On Apple M1 Machines, we need to determine if we need to use the 'x86_64' or 'arm64' debugger.
+    if (platformInfo.isMacOS())
+    {
+        // 'targetArchitecture' is specified in launch.json configuration, use that.
+        if (launchJsonTargetArchitecture)
+        {
+            if (launchJsonTargetArchitecture !== "x86_64" && launchJsonTargetArchitecture !== "arm64")
+            {
+                throw new Error(`The value '${launchJsonTargetArchitecture}' for 'targetArchitecture' in launch configuraiton is invalid. Expected 'x86_64' or 'arm64'.`);
+            }
+            targetArchitecture = launchJsonTargetArchitecture;
+        }
+        else if (dotnetInfo)
+        {
+            // Find which targetArchitecture to use based on SDK Version or RID.
```

Comment:
Do we only want to do the `dotnet --info` path if we have a `.dll`? If we are launching an exe do we want to run `file` on the launch.json `program`?

Reply:
I think that depends on how fancy you want to get (and also how fast `file` is). Personally I think it is reasonable to use the new property you added.